### DISCUSSION TITLE
change tool version install directory

### DISCUSSION THREAD

**Discussion by herda05** [jdx/mise#2366] [Q&A]
Is there an easy way to overwrite the default tool install directory? [According to the doc](https://mise.jdx.dev/cli/#mise-install-options-tool-version), mise install python@3.12.3 will install python into:

> Installs a tool version to `~/.local/share/mise/installs/<PLUGIN>/<VERSION>`

Is there a way to get mise to install this into say /opt/global or some custom path?

**Comment by herda05** [comment]
Digging through the docs, [this seems to be what I'm looking for](https://docs.python.org/3/library/venv.html#how-venvs-work):

> MISE_DATA_DIR
Default: ~/.local/share/mise or $XDG_DATA_HOME/mise
> 
> This is the directory where mise stores plugins and tool installs. These are not supposed to be shared across machines.

So it looks like by putting MISE_DATA_DIR into ~/.config/mise/config.toml, that install will be put into a globally accessible directory that others can then point to as the python binary.

For instance, now when ansible runs, it can use /opt/foo/python in the ansible_python_interpreter variable to ensure a stanardized python deployment across servers.

**Reply by syhol** [reply]
Nice, did this work well? Were there any more issues you encountered with this approach?

**Comment by Marukome0743** [comment]
For current mise versions, the most specific way to change the tool installation directory is to set `MISE_INSTALLS_DIR` in the process environment before mise starts:

```sh
export MISE_INSTALLS_DIR=/opt/global
mise install python@3.12.3
```

`MISE_DATA_DIR` can also relocate mise’s entire data directory, including installs and other mise-managed data, whereas `MISE_INSTALLS_DIR` changes only the installation directory.

One important clarification: `MISE_DATA_DIR` and `MISE_INSTALLS_DIR` should not be configured under `[env]` in `mise.toml`. mise reads these variables during process startup, before it evaluates the config environment. Setting them in `[env]` can make installation and later tool lookup use different directories.

The variable also needs to remain set for later mise commands and shim invocations:

```sh
export MISE_INSTALLS_DIR=/opt/global
```

If `/opt/global` is shared by multiple users, its ownership and permissions must be managed separately. mise does not coordinate concurrent multi-user writes to a shared installation directory, so a centrally managed, read-only installation is generally safer than allowing every user to modify it.

See the current directory documentation: https://mise.en.dev/directories.html

*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*
